src/namingService.ts

```diff
@@ -0,0 +1,13 @@
+import { SourceDefinition } from './types';
+
+export default abstract class NamingService {
+  abstract normalizeSourceDefinition(
```

Comment:
Use `protected` method access modified. We definitelly don't want anyone to call it like `namicorn.ens. normalizeSourceDefinition`

Reply:
Is that consider being just a bad practice or there are bigger reasons for that?